doc/network-upgrade.md

```diff
@@ -0,0 +1,137 @@
+
+- Impact of the upgrade.
+  - Does the new release contain any breaking changes that may require additional contingency plans to be formulated.  Will the update impact the availability of the network for any period of time, or is the upgrade expected to be a typical rolling upgrade with no impact on the network's availability?
+
+    Some upgrades, such as those that contain breaking changes or that require migration scripts require the network upgrade to occur on all nodes of the network simultaneously to avoid any ledger corruption or consensus issues.  During such upgrades the ledger will be completely unavailable for the period of time the upgrade is occurring.
+
+- Pre or post requisites
+
+    Are there any activities that need to be performed on the nodes before an upgrade can commence or after an upgrade is complete?  For example: Do new repository keys need to be registered before the upgrade can be successfully executed on the nodes?
+
+    Such activities can impact the upgrade schedule as they require Steward involvement and resources to be carried out.
+
+- The number of Trustee signatures required to write the `pool-upgrade` transaction to the ledger.
+
+## Performing the upgrade
+
+[Hyperledger Indy Node](https://github.com/hyperledger/indy-node) has a builtin upgrade feature.  A named `pool-upgrade` transaction is written to the `config` ledger of the network.  This transaction is then synchronized with all the nodes on the network and the defined upgrade schedule is registered with each node.  From this point on the nodes manage their own upgrades and report the success or failure to the `config` ledger.  In the case an upgrade fails on a node, it will be retried every 15 minutes until it either succeeds or is canceled.
+
+In the case an upgrade needs to be canceled a matching cancel transaction can be written to the network.  This will cancel any pending node upgrades, or stop a node from retrying a failing upgrade.
+
+The transaction is generated and later written to the ledger using the indy-cli(-rs).
+
+## Generating the `pool-upgrade` transactions
+
+When generating the `pool-upgrade` transaction you should always generate a matching cancel transaction in case it's needed.  Both transactions require Trustee signatures to be written to the ledger.  Generating the cancel transaction at the same time avoids any delays associated with the signing process in the case the upgrade needs to be canceled.
+
+You will need to generate a different set of `pool-upgrade` and cancel commands for each ledger and get each set signed by the appropriate trustees.
+
+The command to generate the `pool-upgrade` command takes the following form:
+
+```
+ledger pool-upgrade name=<UniqueUpgradeName> package=<PackageToUpgrade> version=<UpgradeToPackageVersion> action=start timeout=15 sha256=<PackageSha> schedule=<UpgradeSchedule> sign=false send=false
+```
+
+The command to generate the matching cancel command takes the following form:
```

Comment:
Nothing.  You end up simply not writing the cancel transaction to the ledger if the upgrade is successful.  You generate the cancel transaction in case you need it.  If you don't need it you never write it to the ledger, and it remains unused.  If it was ever inadvertently used in the future it would have zero affect, since the upgrade has already been completed.

Reply:
Can we add  a note to say,  "Cancel transactions are not generally required to write to the ledger unless a special scenario raises to revert the release schedule."

Reply:
I added wording on line 46 to drive home the statement on line 40.